File: Chapter03/3.3.5 DIP/GoodWay/Coach.cs
namespace Dip.GoodWay
{
    // High-level module (also depends on abstractions)
    public class Coach
    {
        private readonly List<IPlayer> _team;

        // Dependencies are "injected" via the constructor!
        public Coach(List<IPlayer> players)
        {
            _team = players;
        }

        public void ExecuteGamePlan()
        {
            foreach (var player in _team)
            {
                player.PerformAction();
            }
        }
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.3.5 DIP/After/Coach.java
package com.grokkingsoftwarearchitecture.chapter03.dip.after;

import java.util.List;

// High-level module (also depends on abstractions)
public class Coach {
    private final List<Player> team;

    // Dependencies are "injected" via the constructor!
    public Coach(List<Player> players) {
        this.team = players;
    }

    public void executeGamePlan() {
        for (Player player : team) {
            player.performAction();
        }
    }
}